gnome-initial-setup/pages/site/gis-site-search-entry.c

```diff
@@ -0,0 +1,570 @@
+
+  completion = gtk_entry_completion_new ();
+
+  gtk_entry_completion_set_popup_set_width (completion, FALSE);
+  gtk_entry_completion_set_text_column (completion,
+                                        GIS_SITE_SEARCH_ENTRY_COL_DISPLAY_NAME);
+  gtk_entry_completion_set_match_func (completion, matcher, NULL, NULL);
+  gtk_entry_completion_set_inline_completion (completion, TRUE);
+
+  g_signal_connect (completion, "match-selected",
+                    G_CALLBACK (match_selected), entry);
+
+  gtk_entry_set_completion (GTK_ENTRY (entry), completion);
+  g_object_unref (completion);
+
+  g_signal_connect (entry, "changed",
+                    G_CALLBACK (entry_changed), NULL);
+}
+
+static void
+finalize (GObject *object)
+{
+  GisSiteSearchEntryPrivate *priv = gis_site_search_entry_get_instance_private (
+    GIS_SITE_SEARCH_ENTRY (object));
+
+  if (priv->site)
+    g_object_unref (priv->site);
+  if (priv->sites)
+    g_ptr_array_unref (priv->sites);
+  if (priv->model)
+    g_object_unref (priv->model);
+
+  G_OBJECT_CLASS (gis_site_search_entry_parent_class)->finalize (object);
+}
+
+static void
+dispose (GObject *object)
+{
+  GisSiteSearchEntryPrivate *priv = gis_site_search_entry_get_instance_private (
+    GIS_SITE_SEARCH_ENTRY (object));
+
+  G_OBJECT_CLASS (gis_site_search_entry_parent_class)->dispose (object);
+}
+
+static void
+constructed (GObject *object)
+{
+  G_OBJECT_CLASS (gis_site_search_entry_parent_class)->constructed (object);
+}
+
+static void
+gis_site_search_entry_class_init (
+        GisSiteSearchEntryClass *site_search_entry_class)
+{
+  GObjectClass *object_class = G_OBJECT_CLASS (site_search_entry_class);
+
+  object_class->constructed = constructed;
+  object_class->finalize = finalize;
+  object_class->set_property = set_property;
+  object_class->get_property = get_property;
+  object_class->dispose = dispose;
+
+  /* properties */
+  g_object_class_install_property (
+      object_class, PROP_SITES_AVAILABLE,
+      g_param_spec_boxed (GIS_SITE_SEARCH_ENTRY_PROP_SITES_AVAILABLE,
+                          "Full site list",
+                          "The array of sites available",
+                          G_TYPE_PTR_ARRAY,
+                          G_PARAM_READWRITE));
+
+  g_object_class_install_property (
+      object_class, PROP_SITE,
+      g_param_spec_object (GIS_SITE_SEARCH_ENTRY_PROP_SITE,
+                           "Site",
+                           "The selected site",
+                           GIS_TYPE_SITE,
+                           G_PARAM_READWRITE));
+}
+
+static void
+set_property_sites_available (GisSiteSearchEntry *self,
+                              const GValue       *value)
+{
+  GisSiteSearchEntryPrivate *priv = gis_site_search_entry_get_instance_private (
+    self);
+  GtkListStore *store = NULL;
+  GtkEntryCompletion *completion;
+
+  store = gtk_list_store_new (3, G_TYPE_STRING, GIS_TYPE_SITE, G_TYPE_STRING);
+
+  priv->sites = g_value_dup_boxed (value);
```

Comment:
Unless the property is set as `CONSTRUCT_ONLY`, this will always leak the previous values.

Reply:
This comment has not been addressed from the previous round.